app/controllers/concerns/course/forum/topics_subscribe_concern.rb

```diff
@@ -0,0 +1,35 @@
+module Course::Forum::TopicsSubscribeConcern
+  extend ActiveSupport::Concern
+
+  def subscribe
+    if subscribe_or_unsubscribe
+      redirect_to course_forum_topic_path(current_course, @forum, @topic),
+                  success: t('course.forum.topics.subscribe.success',
+                             state: subscribed_state, title: @topic.title)
+    else
+      redirect_to course_forum_topic_path(current_course, @forum, @topic),
+                  danger: t('course.forum.topics.subscribe.failure',
+                            state: subscribed_state, title: @topic.title)
+    end
+  end
+
+  private
+
+  def subscribed_state
+    if params[:option] == 'subscribe'
+      t('course.forum.topics.subscribe.state')
+    elsif params[:option] == 'unsubscribe'
+      t('course.forum.topics.unsubscribe.state')
+    end
+  end
+
+  def subscribe_or_unsubscribe
+    if params[:option] == 'subscribe'
+      @topic.subscriptions.create(user: current_user)
+    elsif params[:option] == 'unsubscribe'
+      @topic.subscriptions.where(user: current_user).destroy_all
+    else
+      fail ArgumentError
```

Comment:
Fail with a HTTP 400?

Why is the other actions using true/false but this using subscribe/unsubscribe?